internal/server/singleprocess/state/application.go

```diff
@@ -242,18 +242,14 @@ func (s *State) appPollPeek(
 
 	rec := raw.(*projectIndexRecord)
 	if rec.AppStatusNextPoll.IsZero() {
-		// This happens if this applications poller hasn't been switched on
+		// This happens if this application's poller hasn't been switched on
 		return nil, time.Time{}, nil
 	}
 
 	var result *pb.Project
-	err = s.db.View(func(dbTxn *bolt.Tx) error {
-		var err error
-		result, err = s.projectGet(dbTxn, memTxn, &pb.Ref_Project{
-			Project: rec.Id,
-		})
 
-		return err
+	result, err = s.projectGet(dbTxn, memTxn, &pb.Ref_Project{
+		Project: rec.Id,
 	})
```

Comment:
I _think_ that this fixes an intermittent deadlock I was seeing in the test, but I'm not entirely sure why. In any case, this seems like the right thing to be doing.

Reply:
Oh yes. Lowercase functions should not be creating transactions at all, they assume they're in a transaction already. Only exported functions setup transactions.